src/Symfony/Bridge/Doctrine/Tests/Form/ChoiceList/DoctrineChoiceLoaderTest.php

```diff
@@ -202,10 +209,10 @@ public function testLoadValuesForChoicesDoesNotLoadIfSingleIntId()
             ->with($this->obj2)
             ->willReturn('2');
 
-        $this->assertSame(['2'], $loader->loadValuesForChoices([$this->obj2]));
+        $this->assertSame(['2'], $loader->loadValuesForChoices([$this->obj2], [$this->idReader, 'getIdValue']));
```

Comment:
This change was wrong, I reverted it

Reply:
In fact this is more tricky, by extending the new `AbstractChoiceLoader` the doctrine one should not handle optimisation for getting the values anymore, since the callback is always defined when this is the case, that's why `EntityType` integration tests were still passing but not this one.

So I suggest to either modify this test as I did first, or add some deprecation as I did in another commit I pushed here before simplifying again the `DoctrineChoiceLoader` in 6.0.

WDYT? ping @xabbuh @stof